### ISSUE TITLE
Add RandomSFX Support

### ISSUE THREAD

**Issue by ariadarkkkis** [maoxs2/GxMiner#2]
Hi, Please add RandomSFX Support to miner.

Here are the links:
https://github.com/safex/safexcore
https://github.com/safex/RandomSFX

I guess they have Go support aswell:
https://github.com/safex/gosafex

BTW, Can you make one miner for all algorithms. Like choose algorithm to mine like other miners.

**Comment by maoxs2** [comment]
Added https://github.com/maoxs2/GxMiner/releases/download/v0.1.16/random-sfx.zip

**Comment by ariadarkkkis** [comment]
any source code available?

**Comment by ariadarkkkis** [comment]
@maoxs2 It is submitting invalid results to pool

**Comment by ariadarkkkis** [comment]
`09:08:40 Pool: pool.safexnews.net:4411
09:08:40 Init Job: ID: 32707A108C9C3 Target: 45E7B272F6087 Seed: 13694D957873863
E
09:08:40 Dataset initializing
09:08:41 Dataset initialization finished, start mining
09:08:56 Submitting Sol: 4CC5AAAA for Job 32707A108C9C3 Result: 13D568F6C31CD
09:08:56 Rejected share: invalid result
09:08:59 TOTAL:1.43kh/s SHARES:(0/1)
09:09:15 Submitting Sol: 78440000 for Job 32707A108C9C3 Result: B34526004AFB
09:09:15 Rejected share: invalid result
09:09:19 TOTAL:1.71kh/s SHARES:(0/2)`

**Comment by maoxs2** [comment]
```bash
git clone --recursive https://github.com/maoxs2/GxMiner 
cd GxMiner
./build.sh random-sfx
```

Source code are https://github.com/maoxs2/GxMiner https://github.com/maoxs2/go-randomx and https://github.com/maoxs2/RandomX (branch random-sfx)

It seems that the master branch of https://github.com/safex/RandomSFX/tree/master is the origin random-x, not random-sfx... I had a mistake.

Maybe the SFX-v1.1.6 branch https://github.com/safex/RandomSFX/tree/SFX-v1.1.6

**Comment by maoxs2** [comment]
Binary has been updated with new random-x code https://github.com/safex/RandomSFX/tree/SFX-v1.1.6

**Comment by ariadarkkkis** [comment]
Great its working now, I will submit hashrate difference between this miner and xmrig miner in few minutes. Thanks

**Comment by maoxs2** [comment]
The latest xmrig(v5.4.0) shall be faster. The speed of GxMiner is seriously depending on https://github.com/tevador/RandomX, but now it is obviously lacking tons of optimizations (less commits) comparing to xmrig.

**Comment by ariadarkkkis** [comment]
Gxminer RandomSFX hashrate: 1.95KH/s
Xmrig RandomSFX hashrate: 2.06KH/s

**Comment by ariadarkkkis** [comment]
Hashrate is very good comparing that this is a Go miner and new, comparing to Xmrig which is C++ and old miner ( by old I mean it has more updates and fixed and optimizations ). Very impressive work. Im looking forward to use it.

**Comment by ariadarkkkis** [comment]
@maoxs2 any fixes on API? `/hashrates` showing empty json string.

**Comment by maoxs2** [comment]
fixed on 1cadf0fcece86ef5e12c4268df5d691895ee1b01
